src/Jsontracer/JsonTracerViewerPanel.ts

```diff
@@ -0,0 +1,126 @@
+
+  public static register(context: vscode.ExtensionContext): void {
+    const provider = new JsonTracerViewerPanel(context);
+    const registrations = [
+      vscode.window.registerCustomEditorProvider(JsonTracerViewerPanel.viewType, provider, {
+        webviewOptions: {
+          retainContextWhenHidden: true,
+        },
+      })
+      // Add command registration here
+    ];
+
+    registrations.forEach(disposable => context.subscriptions.push(disposable));
+  };
+
+  constructor(private readonly context: vscode.ExtensionContext) {}
+
+  public async resolveCustomTextEditor(
+      document: vscode.TextDocument, webviewPanel: vscode.WebviewPanel,
+      _token: vscode.CancellationToken): Promise<void> {
+    await this.initWebview(document, webviewPanel.webview);
+    this.initWebviewPanel(document, webviewPanel);
+    this.updateWebview(document, webviewPanel.webview);
+  };
+
+  private async initWebview(document: vscode.TextDocument, webview: vscode.Webview): Promise<void> {
+    webview.options = {
+      enableScripts: true,
+    };
+
+    const nonce = getNonce();
+    const scriptUri =
+        getUri(webview, this.context.extensionUri, ['media', 'Jsontracer', 'index.js']);
+    const styleUri =
+        getUri(webview, this.context.extensionUri, ['media', 'Jsontracer', 'style.css']);
+    const htmlUri = vscode.Uri.joinPath(this.context.extensionUri, 'media/Jsontracer/index.html');
+    let html = Buffer.from(await vscode.workspace.fs.readFile(htmlUri)).toString();
+    html = html.replace(/\${nonce}/g, `${nonce}`);
+    html = html.replace(/\${webview.cspSource}/g, `${webview.cspSource}`);
+    html = html.replace(/\${scriptUri}/g, `${scriptUri}`);
+    html = html.replace(/\${styleUri}/g, `${styleUri}`);
+    webview.html = html;
+  }
+
+  private initWebviewPanel(document: vscode.TextDocument, webviewPanel: vscode.WebviewPanel): void {
+    vscode.commands.executeCommand('setContext', JsonTracerViewerPanel.viewType, true);
+
+    const changeDocumentSubscription = vscode.workspace.onDidChangeTextDocument(e => {
+      if (e.contentChanges.length > 0 && e.document.uri.toString() === document.uri.toString()) {
+        this.updateWebview(document, webviewPanel.webview);
+      }
+    });
+
+    webviewPanel.onDidChangeViewState(e => {
+      vscode.commands.executeCommand(
+          'setContext', JsonTracerViewerPanel.viewType, webviewPanel.visible);
+    }, null, this._disposables);
+
+    webviewPanel.onDidDispose(() => {
+      changeDocumentSubscription.dispose();
+      while (this._disposables.length) {
+        const x = this._disposables.pop();
+        if (x) {
+          x.dispose();
+        }
+      }
+      vscode.commands.executeCommand('setContext', JsonTracerViewerPanel.viewType, false);
+    });
+  }
+
+  private updateWebview(document: vscode.TextDocument, webview: vscode.Webview): void {
+    webview.postMessage({type: 'load', content: JSON.parse(document.getText()).traceEvents});
```

Comment:
(optional, maybe TODO later)

`content: JSON.parse(document.getText()).traceEvents`
I think this will trigger some javscript errors, how about handling the case?